### DISCUSSION TITLE
AFP FileSharing Mac OSX 10.5 Leopard with OS 13 Ventura via Netatalk ?

### DISCUSSION THREAD

**Discussion by DumpfbackeII** [Netatalk/netatalk#573] [Q&A]
Hello, I would like to access the data on my new Mac on OS13 Ventura from my very old Mac with OSX 10.5 Leopard.

I installed netatalk on Ventura using Macports ( https://ports.macports.org/port/netatalk/ ). This has worked so far. However, I have not found anywhere what exactly the config file should look like?

The connection from 10.5 to Ventura via afp://"IP-Adress" should work in principle now, but I fail at the login. I get the login window, but I can't log in.

I have tried different variations bus does someone maybe know how a config file should look like on the Mac, so that it works ?
Or / and is there maybe anything else to do so that I can login ?

I have to say, I've been working with Mac for a long time, so I know my way around quite well, but I'm not a network engineer...

For your interest: Connecting from OSX 10.5 Leopard via vnc://"IP-Adress" to OS13 Ventura still works ! But doesn't really help for file Sharing...

-------------------------------

Netatalk on Ventura:

Terminal > afpd -V

afpd 2.2.10 - Apple Filing Protocol (AFP) daemon of Netatalk

This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free Software
Foundation; either version 2 of the License, or (at your option) any later
version. Please see the file COPYING for further information and details.

afpd has been compiled with support for these features:

AFP versions: 2.2 3.0 3.1 3.2 3.3
DDP(AppleTalk) Support: No
CNID backends: dbd last tdb
SLP support: No
Zeroconf support: Avahi
TCP wrappers support: No
Quota support: No
Admin group support: Yes
Valid shell checks: Yes
cracklib support: No
Dropbox kludge: No
Force volume uid/gid: No
EA support: ad | sys
ACL support: No
LDAP support: Yes

afpd.conf: /opt/local/etc/netatalk/afpd.conf
AppleVolumes.system: /opt/local/etc/netatalk/AppleVolumes.system
AppleVolumes.default: /opt/local/etc/netatalk/AppleVolumes.default
afp_signature.conf: /opt/local/etc/netatalk/afp_signature.conf
afp_voluuid.conf: /opt/local/etc/netatalk/afp_voluuid.conf
afp_ldap.conf: /opt/local/etc/netatalk/afp_ldap.conf
UAM search path: /opt/local/etc/netatalk/uams/
Server messages path: /opt/local/etc/netatalk/msg/
lockfile: /var/run/afpd

**Comment by rdmark** [comment]
What error message exactly are you getting when attempting to authenticate from the Leopard system?

The most likely cause would be the lack of a compatible UAM in your afpd.conf -- i.e. the Leopard system can't see a way to authenticate with the netatalk server.

One thing you can try is to disable the DHX UAM and activate the ClearText UAM. There's a known issue with DHX and OpenSSL 3.0: https://github.com/Netatalk/netatalk/issues/358

For details on how to configure UAMs see the manual page: https://netatalk.sourceforge.io/2.2/htmldocs/afpd.conf.5.html

If you want pointers on what exactly to put in your afpd.conf, please share the contents of your afpd.conf first. :)

**Comment by DumpfbackeII** [comment]
hi,

thanks, but as I said, iam not am expert in this ...
The UAMs manual page is like to speak French to me :)
i hoped someone did this before and can give me an step per step help how to do...


my conf-file:

[Global]
afpstats = yes
#ldap auth method = simple
uam list = DHX2
[Homes]
basedir regex = /home


my error ;)
![IMG_4555](https://github.com/Netatalk/netatalk/assets/152258049/e8c9dcf0-95c3-4922-9ea5-10eaf47b7e3f)

**Reply by rdmark** [reply]
Ok so I see one problem here: The config file you shared is for netatalk 3.x, while the version you showed being installed above is netatalk 2.x. The two versions have very different config file formats, and have have slightly different naming: `afp.conf` for netatalk 3.x and `afpd.conf` for netatalk 2.x... 

You should not run both at the same time since they serve their AFP server on the same port by default. Can you please make sure you don't have netatalk 3.x running already before starting netatalk 2.x?

And then please share the contents of your netatalk 2.x `afpd.conf` file.

**Comment by DumpfbackeII** [comment]
i installed Netatalk with macports - see the screenshot - this seems to be version 2.

I have not installed an other version ?

the afpd.conf file was blank after installation -> so I wrote the config file by myself with infos I found in the internet... there aim not wondering when its wrong ;)

do i have to install version 3 ?

or change the config file to ?

<img width="1369" alt="Bildschirmfoto 2023-12-03 um 15 41 41" src="https://github.com/Netatalk/netatalk/assets/152258049/476c363c-42c0-4ce7-b2e3-f7e58e0961a7">

**Reply by rdmark** [reply]
> the afpd.conf file was blank after installation -> so I wrote the config file by myself with infos I found in the internet... there aim not wondering when its wrong ;)

It shouldn't have been blank. That's either a problem with MacPorts making a broken netatalk installation, or that you looked in the wrong place for adpf.conf... Are all the other netatalk conf files blank too, such as `AppleVolumes.default`?

> do i have to install version 3 ?

For your usecase, either version will serve you well. Version 2.x is required for AppleTalk neworking (older Classic Mac OS), while 3.x has Spotlight support and AFP 3.4 conformance.

**Comment by DumpfbackeII** [comment]
my config file ist under:
etc/Netatalk/afpd
iam not sure, but I think I also crated the folder Netatalk !

I havent found a AppleVolumes.default 
where should this be ?

**Reply by rdmark** [reply]
If you look at your very first post, the locations of the config files are right there. Here's a snippet:

afpd.conf: /opt/local/etc/netatalk/afpd.conf

AppleVolumes.default: /opt/local/etc/netatalk/AppleVolumes.default

So you've been working in the wrong directory there. 
Please share the contents of the above files.

**Comment by DumpfbackeII** [comment]
```
okay, there it is :)

the afpd.conf:

#
# CONFIGURATION FOR AFPD (Netatalk 2.x)
#
# Each single line defines a virtual server that should be available.
# Though, using "\" character, newline escaping is supported.
# Empty lines and lines beginning with `#' are ignored.
# Options in this file will override both compiled-in defaults
# and command line options.
#


#
# Format:
#  - [options]               to specify options for the default server
#  "Server name" [options]   to specify an additional server
#


#
# Some examples:
#
#       The simplest case is to not have an afpd.conf.
#
#       4 servers w/ names server1-3 and one w/ the hostname. Servers
#       1-3 get routed to different ports with server3 being bound
#       specifically to address 192.168.1.3
#
#           -
#           server1 -port 12000
#           server2 -port 12001
#           server3 -port 12002 -ipaddr 192.168.1.3
#
#       A dedicated guest server, a user server, and a special
#       AppleTalk-only server:
#
#           "Guest Server" -uamlist uams_guest.so \
#                   -loginmesg "Welcome guest! I'm a public server."
#           "User Server" -uamlist uams_dhx2.so -port 12000
#           "special" -ddp -notcp -defaultvol <path> -systemvol <path>
#


# default:
# - -transall -uamlist uams_dhx.so,uams_dhx2.so

----------------------------
and the AppleVolumes.default:

# Netatalk 2.x afp volume cofiguration

#
# volume format:
# :DEFAULT: [all of the default options except volume name]
# path [name] [casefold:x] [options:z,l,j] \
#   [allow:a,@b,c,d] [deny:a,@b,c,d] [dbpath:path] [password:p] \
#   [rwlist:a,@b,c,d] [rolist:a,@b,c,d] [limitsize:value in bytes] \
#   [preexec:cmd] [root_preexec:cmd] [postexec:cmd]  [root_postexec:cmd] \
#   [allowed_hosts:IPv4 address[/IPv4 netmask bits]] \
#   [denied_hosts:IPv4 address[/IPv4 netmask bits]] \
#   ... more, see below ...
#   
# name:      volume name. it can't include the ':' character
#

# The line below sets some DEFAULT, starting with Netatalk 2.1.
:DEFAULT: options:upriv,usedots

# The "~" below indicates that Home directories are visible by default.
# If you do not wish to have people accessing their Home directories,
# please put a pound sign in front of the tilde or delete it.
~

# End of File

```

**Reply by rdmark** [reply]
Good, so you have the default config files in place. Then try adding this line to the end of your afpd.conf and restart afpd.service

```
- -transall -uamlist uams_clrtxt.so,uams_dhx2.so
```

**Comment by DumpfbackeII** [comment]
unfortunately no differenz
I see in "litte snitch" that it wants to connect - of corse I allow the connection ;)
But the login still doesn't work... 

I tried
afp:// IP-Adress
afp:// ComputerName-User.local

<img width="804" alt="Bildschirmfoto 2023-12-06 um 18 23 16" src="https://github.com/Netatalk/netatalk/assets/152258049/f631119f-cc69-4927-8631-cc9f1a340a24">

**Reply by rdmark** [reply]
Can you turn off your firewall and see if you get any further?

As per the output of `afpd -V` your netatalk server should be broadcasting itself with zeroconf so your Bonjour enabled OSX system should be seeing the server automatically in the Network tray if the network traffic works properly. There should not be any need to input IP addresses to connect.

**Comment by DumpfbackeII** [comment]
_Can you turn off your firewall and see if you get any further?_
I tried this already, no...

_OSX system should be seeing the server automatically in the Network tray_
if you mean in the finder ? yes and no, I see them, but if I try to connect there comes only an error that it can not connect. I thought this connection ist per SMB, which also isn't working anymore...


i will make a something else. I bought an old used Mac mini, you can get them very cheap... change the HDD for more space and install El Capitan on it - so I can put all my stuff on the Mac mini and work from the old and the new Mac because both system can access the system in  the middle ;) Also I cam simple make an backup with Time Machine and an extern drive. 

thanks for your help anyway !!!
